### PR TITLE
Add GTEST_QUIET option

### LINKED ISSUE #358: Quiet output
```
There is no flag to pass to InitGoogleTest that results in minimal console 
output. Such a flag would tell gtest to only output errors from failed tests, 
the number of tests passed, and the total time elapsed. This flag is necessary 
if one wants to compile all tests into a executable to be run as a post-build 
step.
```

Original issue reported on code.google.com by `mahsm...@gmail.com` on 4 Mar 2011 at 8:41

### PR DESCRIPTION
**Pull request by calumr**
Only prints failures and a total pass count. This is useful for CI logs, which are more useful if they just have the failures instead of 100s or 1000s of lines of "PASSED" and the one failure buried in the middle. 

Fixes #358, which was closed but seems to have a few fans...

### CODE REVIEW

**Comment by calumr** [comment]
@googlebot I signed it!

File: googletest/docs/advanced.md [hunk 1]
```diff
@@ -2252,6 +2252,12 @@ disable colors, or let googletest decide. When the value is `auto`, googletest
 will use colors if and only if the output goes to a terminal and (on non-Windows
 platforms) the `TERM` environment variable is set to `xterm` or `xterm-color`.
 
+#### Suppressing test passes
+
+By default, googletest prints 1 line of output for each test, indicating if it
+passed or failed. T to show only test failures, run the test program with
```
[the remark this answers is not in the document]
Comment (calumr) [reply googletest/docs/advanced.md]: Thanks, fixed in the last push.

**Comment by calumr** [comment]
You were totally right about adding another TestEventListener - it's much cleaner this way. There is some code duplication though, I can understand if you want me to clean that up.

I went a little further and removed the PrintFailedTests calls too, they are just duplicating the information printed from OnTestEnd.

**Comment by lishuai87** [comment]
ctest has option `--output-on-failure`, why gtest not support it?

https://cmake.org/cmake/help/latest/manual/ctest.1.html#options

**Pull request merged**